### PR TITLE
feat: move to tap15

### PR DESCRIPTION
**Pull request by zekth**

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -6,19 +6,21 @@ const test = t.test
 const Fastify = require('fastify')
 const fastifyRedis = require('..')
 
-t.beforeEach((done) => {
-  const fastify = Fastify()
+t.beforeEach(() => {
+  return new Promise((resolve) => {
```
Comment (mcollina) [inline test/test.js:10]: please use async/await instead

**Review by mcollina — CHANGES_REQUESTED** [review]

**Comment by mcollina** [comment]
Also this does not bump the tap dependency.

**Comment by zekth** [comment]
Looks like tap 15 coverage behaviour of the cli has changed. Now it defaults to 100

File: test/test.js [hunk 2]
```diff
@@ -6,20 +6,20 @@ const test = t.test
...
-t.beforeEach((done) => {
+t.beforeEach(async () => {
   const fastify = Fastify()
 
   fastify.register(fastifyRedis, {
     host: '127.0.0.1'
   })
 
-  fastify.ready((err) => {
+  fastify.ready(async (err) => {
```
Comment (mcollina) [inline test/test.js:16]:
```suggestion
  await fastify.ready()
```

File: test/test.js [hunk 3]
```diff
@@ -6,20 +6,20 @@ const test = t.test
...
     t.error(err)
 
-    fastify.redis.flushall(() => {
-      fastify.close()
-      done()
-    })
+    await fastify.redis.flushall()
+      .then(() => {
```
Comment (mcollina) [inline test/test.js:20]: remove the `.then()`

File: test/test.js [hunk 4]
```diff
@@ -6,20 +6,20 @@ const test = t.test
...
+        fastify.close()
```
Comment (mcollina) [inline test/test.js:21]:
```suggestion
        await fastify.close()
```

**Review by mcollina — CHANGES_REQUESTED** [review]
The deprecations must be fixed

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**